src/resource.rs

```diff
@@ -22,17 +22,20 @@ pub use type_def::{TypeDef, TypeDefTag};
 
 /// An error that can occur in computing the signature of a node.
 /// TODO: decide on failure modes
-#[derive(Debug, Clone, Error, PartialEq, Eq)]
+#[derive(Debug, Clone, Error, PartialEq)]
```

Comment:
I had to drop the `Eq` here because we are reusing `ConstTypeError`, and according to its definition a `ConstTypeError` could contain a ValueCheckFail with an arbitrary (non-Eq) ConstValue. *BUT* of course in practice here we are checking only TypeParams-vs-TypeArgs, so we'll never see such a ValueCheckFail (we'll only get TypeArgCheckFails).

So, the right thing to do would be to *parameterize* `ConstTypeError` by a `ValueOfType` and then combine `ValueCheckFail` with `TypeArgCheckFail` (using the `T: ValueOfType`). I haven't experimented with this, but will have a go...